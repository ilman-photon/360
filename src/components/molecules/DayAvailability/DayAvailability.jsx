import Box from "@mui/material/Box";
import React, { useEffect, useState } from "react";
import { StyledButton } from "../../atoms/Button/button";
import styles from "./styles.module.scss";
import { Divider, Typography } from "@mui/material";
import constants from "../../../utils/constants";
import ArrowBackIosIcon from "@mui/icons-material/ArrowBackIos";
import ArrowForwardIosIcon from "@mui/icons-material/ArrowForwardIos";
import {
  parseDateWeekList,
  parseScheduleDataWeekOverlay,
  timeInWeekLabel,
} from "../../../utils/appointment";

export const buttonSchedule = (
  label = "",
  idx,
  OnDayClicked = () => {
    // This is intended
  },
  date = "",
  isScheduleAvailability = false
) => {
  const isNextAvailabilityLabel =
    !isScheduleAvailability ||
    (isScheduleAvailability && label.indexOf("Next availability is") < 0);
  const dateTime = !isNextAvailabilityLabel
    ? new Date(`${date} ${label.toUpperCase().replace(/(AM|PM)/, " $1")}`)
    : "";
  return (
    <Box
      key={idx}
      sx={{ width: !isScheduleAvailability ? "78px" : "100%" }}
      className={styles.scheduleBtnWarpper}
    >
      <StyledButton
        theme={constants.PATIENT}
        mode={constants.PRIMARY}
        size={constants.SMALL}
        gradient={false}
        className={
          !isScheduleAvailability ? styles.scheduleBtn : styles.scheduleAvailBtn
        }
        onClick={() => {
          if (isNextAvailabilityLabel) {
            OnDayClicked(dateTime);
          }
        }}
      >
        {label}
      </StyledButton>
    </Box>
  );
};

export const DayAvailability = ({
  rangeDate = {
    startDate: "",
    endDate: "",
  },
  scheduleData = {},
  isDesktop = false,
  OnDayClicked = () => {
    // This is intended
  },
}) => {
  const [schedule, setSchedule] = useState({});
  const [timeInWeek, setTimeInWeek] = useState("");
  const [dateWeekList, setDateWeekList] = useState([]);

  useEffect(() => {
    const scheduleParse = parseScheduleDataWeekOverlay(scheduleData);
    if (scheduleParse) {
      setSchedule(scheduleParse);
      setDateWeekList(parseDateWeekList(scheduleData));
    }

    setTimeInWeek(timeInWeekLabel(rangeDate.startDate, rangeDate.endDate));
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [scheduleData]);

  function renderScheduleData() {
    let renderUI = [];
    for (const [index, [key, value]] of Object.entries(
      Object.entries(schedule)
    )) {
      if (value && value.length > 0) {
        renderUI.push(
          <Box className={styles.scheduleContainer} key={index}>
            <Typography className={styles.scheduleTitle}>{key}</Typography>
            {renderTimeSchedule(value, index)}
          </Box>
        );
      } else {
        renderUI.push(
          <Box
            key={index}
            className={[
              styles.scheduleContainer,
              styles.noScheduleContainer,
            ].join(" ")}
          >
            <Typography className={styles.scheduleTitle}>{key}</Typography>
            <Typography className={styles.noSchedule}>
              No availability
            </Typography>
          </Box>
        );
      }
    }
    return renderUI;
  }

  function renderTimeSchedule(value, index) {
    const column = isDesktop ? 5 : 4;
    return (
      <Box
        sx={{
          display: "grid",
          width: isDesktop ? 410 : 330,
          gridTemplateColumns: `repeat(${column}, 1fr)`,
          gap: "4px",
          justifyContent: "center",
          alignContent: "center",
          gridTemplateRows: "auto",
        }}
      >
        {value.map((option, idx) => {
          return buttonSchedule(option, idx, OnDayClicked, dateWeekList[index]);
        })}
      </Box>
    );
  }

  return (
    <Box>
      <Box className={styles.scheduleTimeContainer}>
        <Typography className={styles.scheduleTimeTitle}>
          {timeInWeek}
        </Typography>
        <Box className={styles.iconTimeContainer}>
          <ArrowBackIosIcon className={styles.iconSchedule} />
          <ArrowForwardIosIcon className={styles.iconSchedule} />
        </Box>
      </Box>
      <Divider className={styles.dividerSchedule} />
      {renderScheduleData()}
    </Box>
  );
};
