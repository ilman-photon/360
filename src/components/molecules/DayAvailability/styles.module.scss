@import "../../../../styles/colors.scss";

.scheduleContainer,
.noScheduleContainer {
  .scheduleTitle {
    font-family: "Museo Sans", sans-serif;
    font-style: normal;
    font-weight: 500;
    font-size: 15px;
    line-height: 24px;
    color: $darkGreen;
    letter-spacing: 0.0016em;
    width: 106px;
  }

  .noSchedule {
    width: 106.5px;
    font-family: "Museo Sans", sans-serif;
    font-style: normal;
    font-weight: 500;
    font-size: 15px;
    line-height: 24px;
    color: #a5a5ac;
    letter-spacing: 0.0016em;
    margin-left: 8px;
  }
}

.noScheduleContainer {
  display: flex;
}

.scheduleBtnWarpper {
  .scheduleBtn {
    width: 78px;
    height: 30px;
    padding: 0;
    background: #007e8f;
    border-radius: 30px;
    margin-top: 8px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    font-family: "Museo Sans", sans-serif;
    font-style: normal;
    font-weight: 500;
    font-size: 14px;
    line-height: 18px;
  }
  .scheduleAvailBtn {
    width: 100%;
    padding: 0;
    height: 30px;
    background: #007e8f;
    border-radius: 30px;
    margin-top: 8px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    font-family: "Museo Sans", sans-serif;
    font-style: normal;
    font-weight: 500;
    font-size: 14px;
    line-height: 18px;
  }
  .scheduleDisableBtn {
    background: #b1b1b1;
    pointer-events: none;
    cursor: not-allowed;
  }
}

.scheduleTimeContainer {
  display: flex;
  gap: 18.71px;
  margin-bottom: 12px;
  position: relative;
  .scheduleTimeTitle {
    font-family: "Museo Sans", sans-serif;
    font-style: normal;
    font-weight: 600;
    font-size: 18px;
    line-height: 26px;
    /* identical to box height, or 144% */
    /* ECP/Primary/DarkGreen */

    color: #003b4a;
    display: flex;
    align-items: center;
    letter-spacing: 0.0016em;
    min-width: 135px;
  }
  .iconTimeContainer {
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    padding: 0px;
    gap: 12px;

    width: 359px;
    height: 26px;
    .iconSchedule {
      width: 24px;
      height: 24px;
      color: $darkGreen;
      cursor: pointer;
      opacity: 1;
    }
    .iconDisableSchedule {
      width: 24px;
      height: 24px;
      color: $darkGreen;
      cursor: not-allowed;
      opacity: 0.3;
    }
  }

  .dividerSchedule {
    margin-top: 12px;
  }
}
