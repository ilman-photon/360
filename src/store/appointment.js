import { faker } from "@faker-js/faker";
import { createSlice } from "@reduxjs/toolkit";

const DEFAULT_PROVIDER_INFO_DATA = {
  providerId: 0,
  name: null,
  address: null,
  rating: null,
  phoneNumber: null,
  distance: null,
  image: null,
  from: null,
  to: null,
  location: {
    latitude: 0,
    longitude: 0,
  },
};

export const DEFAULT_PATIENT_INFO_DATA = {
  name: null,
  firstName: "",
  lastName: "",
  dob: null,
  phoneNumber: null,

  email: "",
  password: "",
  preferredCommunication: "both",
};

const DEFAULT_APPOINTMENT_INFO_DATA = {
  appointmentType: null,
  date: null,
  insuranceCarrier: null,
};

const DEFAULT_USER_SCHEDULE_APPOINTMENT_DATA = {
  providerInfo: DEFAULT_PROVIDER_INFO_DATA,
  patientInfo: DEFAULT_PATIENT_INFO_DATA,
  appointmentInfo: DEFAULT_APPOINTMENT_INFO_DATA,
};

const DEFAULT_FILTER_DATA = {
  date: new Date(),
  location: "",
  insuranceCarrier: "",
  purposeOfVisit: "",
};

const appointmentSlice = createSlice({
  name: "appointment",
  initialState: {
    appointmentSchedule: DEFAULT_USER_SCHEDULE_APPOINTMENT_DATA,
    filterData: DEFAULT_FILTER_DATA,
    isFilterApplied: false,
    providerListData: [],
  },
  reducers: {
    setAppointmentSchedule: (state, { payload }) => {
      state.appointmentSchedule = payload;
    },
    resetAppointmentSchedule: (state) => {
      state.appointmentSchedule = DEFAULT_USER_SCHEDULE_APPOINTMENT_DATA;
      state.isFilterApplied = false;
      state.providerListData = [];
    },
    editAppointmentScheduleData: (state, { payload }) => {
      state.appointmentSchedule[payload.key] = payload.value;
    },
    setDummyAppointmentSchedule: (state) => {
      state.appointmentSchedule = {
        providerInfo: {
          providerId: 0,
          name: "Dr. Sonha Nguyen",
          address: {
            addressLine1: "673 Herzog Locks", // faker.address.streetAddress(),
            addressLine2: "Suite 300", // faker.address.secondaryAddress(),
            city: "New York", // faker.address.city(),
            state: "NY", // faker.address.stateAbbr(),
            zipcode: "53891", // faker.address.zipCode("#####"),
          },
          rating: 5, //faker.helpers.arrayElement[(1, 2, 3, 4, 5)],
          phoneNumber: "(123) 123-4567", // faker.phone.number("(###) ###-####"),
          distance: "10 mi",
          image: faker.image.imageUrl(275, 173),
          from: new Date(),
          to: new Date(),
          location: {
            latitude: 41.881832, // faker.address.latitude(),
            longitude: -87.623177, // faker.address.longitude(),
          },
        },
        patientInfo: DEFAULT_PATIENT_INFO_DATA,
        appointmentInfo: {
          appointmentType: "Eye Exam",
          date: "2022-08-29T11:12:55.367Z",
          insuranceCarrier: "RSUD",
        },
      };
    },
    setFilterData: (state, { payload }) => {
      state.filterData = payload;
      state.appointmentSchedule.appointmentInfo = {
        ...state.appointmentSchedule.appointmentInfo,
        appointmentType: payload.purposeOfVisit,
        insuranceCarrier: payload.insuranceCarrier,
      };
    },
    resetFilterData: (state) => {
      state.filterData = DEFAULT_FILTER_DATA;
    },
    setIsFilterApplied: (state, { payload }) => {
      state.isFilterApplied = payload;
    },
    setProviderListData: (state, { payload }) => {
      state.providerListData = payload;
    },
  },
});

// Action creators are generated for each case reducer function
export const {
  setAppointmentSchedule,
  resetAppointmentSchedule,
  editAppointmentScheduleData,
  setDummyAppointmentSchedule,
  setFilterData,
  resetFilterData,
  setIsFilterApplied,
  setProviderListData,
} = appointmentSlice.actions;

export default appointmentSlice.reducer;
