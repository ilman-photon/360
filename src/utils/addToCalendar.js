import { atcb_action } from "add-to-calendar-button";
import moment from "moment";

export const addToCalendar = (data) => {
  const date = new Date(data.date);
  const momentDateRaw = new moment(date);
  const momentDate = momentDateRaw.tz("America/New_York");
  const startDate = momentDate.format("YYYY-MM-DD");
  const endDate = momentDate.format("YYYY-MM-DD");
  const startTime = momentDate.format("hh:mm");
  const endTime = momentDate.add(1, "hours").format("hh:mm");
  atcb_action({
    startDate,
    endDate,
    startTime,
    endTime,
    name: data.name,
    description: data.description,
    location: data.location,
    options: ["Apple", "Google", "Microsoft365", "Outlook.com", "Yahoo"],
    timeZone: "America/New_York",
    iCalFileName: data.name,
    inline: true,
    trigger: "click",
    background: false,
  });
};
