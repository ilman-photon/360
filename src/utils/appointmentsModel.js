import moment from "moment";

export function appointmentParser(data = {}) {
  const firstname = data.provider?.firstName || "";
  const lastName = data.provider?.lastName ? ` ${data.provider?.lastName}` : "";
  const designation = data.provider?.designation
    ? `, ${data.provider?.designation}`
    : "";
  const name = `${firstname}${lastName}${designation}`;

  const momentDate = new moment(
    `${data.appointmentDate} ${data.appointmentTime}`
  );
  const year = momentDate.format("YYYY");
  return {
    appointmentId: data._id,
    providerInfo: {
      providerId: data.provider?._id,
      name,
      position: data.provider?.address?.name || "",
      address: data.provider?.address || "",
      rating: data.provider?.rating / 2 || "",
      phoneNumber: data.provider?.workPhone || "",
      image: data.provider?.profilePhoto?.digitalAsset || "",
    },
    patientInfo: {
      name: `${data.patient?.firstName || ``}${
        data.patient?.lastName ? ` ${data.patient?.lastName}` : ``
      }`,
      firstname: data.patient?.firstName || ``,
      lastname: data.patient?.lastName || ``,
      dob: data.patient?.dob || ``,
    },
    appointmentInfo: {
      appointmentType: data.appointmentType?.name,
      date: `${data.appointmentDate} ${data.appointmentTime}`,
      insuranceCarrier: data.insurancePayers || [],
      state: data.state,
    },
    year,
  };
}

export function groupBy(list, keyGetter) {
  const map = new Map();
  list.forEach((item) => {
    const key = keyGetter(item);
    if (!map.has(key)) {
      map.set(key, [item]);
    } else {
      map.get(key).push(item);
    }
  });
  return map;
}
