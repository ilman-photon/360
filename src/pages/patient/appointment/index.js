import AppointmentLayout from "../../../components/templates/appointmentLayout";
import { Provider } from "react-redux";
import store from "../../../store/store";
import FilterHeading from "../../../components/molecules/FilterHeading/filterHeading";
import { CircularProgress, Grid, useMediaQuery } from "@mui/material";
import Box from "@mui/material/Box";
import React, { useState } from "react";
import FilterResult from "../../../components/molecules/FilterResult/filterResult";
import GMaps from "../../../components/organisms/Google/Maps/gMaps";
import { useLoadScript } from "@react-google-maps/api";

export async function getStaticProps() {
  return {
    props: {
      googleApiKey: process.env.GOOGLE_API_KEY,
    },
  };
}
export default function Appointment({ googleApiKey }) {
  const isDesktop = useMediaQuery("(min-width: 992px)");
  const [isFilterApplied, setFilterApplied] = useState(false);

  const { isLoaded } = useLoadScript({
    googleMapsApiKey: googleApiKey,
  });

  function onSearchProvider() {
    setFilterApplied(true);
  }

  return (
    <Box display="flex" flex={1}>
      <FilterHeading
        isDesktop={isDesktop}
        onSearchProvider={onSearchProvider}
      />
      {isDesktop && isFilterApplied ? (
        <Grid
          container
          sx={{
            display: "flex",
            flex: 1,
            pt: 17,
            maxHeight: "calc(100vh - 80px)",
            overflow: "hidden",
          }}
        >
          <Box
            sx={{
              m: 3,
              overflow: "auto",
              height: "100%",
            }}
          >
            <FilterResult />
          </Box>
          <Box
            sx={{
              background: "#F4F4F4",
              display: "flex",
              flex: 1,
            }}
          >
            {/* Map */}
            {isLoaded ? <GMaps apiKey={googleApiKey} /> : <CircularProgress />}
          </Box>
        </Grid>
      ) : (
        <></>
      )}
    </Box>
  );
}

Appointment.getLayout = function getLayout(page) {
  return (
    <Provider store={store}>
      <AppointmentLayout currentActivePage={"appointment"}>
        {page}
      </AppointmentLayout>
    </Provider>
  );
};
