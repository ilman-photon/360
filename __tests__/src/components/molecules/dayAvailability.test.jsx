import { fireEvent, render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import { DayAvailability } from "../../../../src/components/molecules/DayAvailability/DayAvailability";
import { parseScheduleDataWeekOverlay } from "../../../../src/utils/appointment";

const provider_data = [
  {
    date: "2022-09-19",
    list: [
      {
        time: "11:30am",
        key: 12222,
      },
    ],
  },
  {
    date: "2022-09-20",
    list: [
      {
        time: "08:00am",
        key: 12223,
      },
      {
        time: "10:30am",
        key: 12224,
      },
      {
        time: "11:00am",
        key: 12225,
      },
      {
        time: "12:00pm",
        key: 12226,
      },
      {
        time: "13:00pm",
        key: 12227,
      },
      {
        time: "14:00pm",
        key: 12228,
      },
    ],
  },
  {
    date: "2022-09-21",
    list: [
      {
        time: "08:30am",
        key: 12229,
      },
      {
        time: "10:30am",
        key: 12230,
      },
      {
        time: "11:30am",
        key: 12231,
      },
      {
        time: "12:00pm",
        key: 12232,
      },
      {
        time: "13:30pm",
        key: 12233,
      },
      {
        time: "14:30pm",
        key: 12234,
      },
      {
        time: "15:30pm",
        key: 12235,
      },
      {
        time: "16:30pm",
        key: 12236,
      },
      ,
    ],
  },
  {
    date: "2022-09-22",
    list: [
      {
        time: "09:30am",
        key: 12237,
      },
      {
        time: "11:00am",
        key: 12238,
      },
    ],
  },
  {
    date: "2022-09-23",
    list: [
      {
        time: "09:30am",
        key: 12239,
      },
    ],
  },
  {
    date: "2022-09-24",
    list: [
      {
        time: "09:30am",
        key: 12240,
      },
    ],
  },
]

describe("DayAvailability Components", () => {
  let container;
  beforeEach(() => {
    const scheduleParse = parseScheduleDataWeekOverlay(provider_data[0]);
    container = render(<DayAvailability scheduleData={provider_data[0]}/>);
  });

  it("DayAvailability render", async() => {
    expect(
      await waitFor(() =>
        container.getByTestId("appointment_all_availability_next_week_button")
      )
    ).toBeInTheDocument();
  });

  it("FilterHeadingFilled render", () => {
    container = render(<DayAvailability isDesktop={true} scheduleData={provider_data[0]}/>);
  });
});
