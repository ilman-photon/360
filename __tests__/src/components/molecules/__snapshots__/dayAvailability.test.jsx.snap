// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayAvailability Components DayAvailability render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="scheduleTimeContainer MuiBox-root css-0"
        >
          <p
            aria-label=""
            class="MuiTypography-root MuiTypography-body1 scheduleTimeTitle css-ahj2mt-MuiTypography-root"
            tabindex="0"
          />
          <div
            class="iconTimeContainer MuiBox-root css-0"
          >
            <button
              aria-label="Navigate to previous week option"
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-a4pt68-MuiButtonBase-root-MuiButton-root"
              role="button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconSchedule css-lw08pq-MuiSvgIcon-root"
                data-testid="appointment_all_availability_previous_week_button"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="Navigate to next week option"
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-a4pt68-MuiButtonBase-root-MuiButton-root"
              role="button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconSchedule css-1c40g22-MuiSvgIcon-root"
                data-testid="appointment_all_availability_next_week_button"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth dividerSchedule css-9mgopn-MuiDivider-root"
        />
        <div
          class="css-1lwbda4-MuiStack-root"
          data-testid="appointment_all_availability_schedule_ontainer"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="scheduleTimeContainer MuiBox-root css-0"
      >
        <p
          aria-label=""
          class="MuiTypography-root MuiTypography-body1 scheduleTimeTitle css-ahj2mt-MuiTypography-root"
          tabindex="0"
        />
        <div
          class="iconTimeContainer MuiBox-root css-0"
        >
          <button
            aria-label="Navigate to previous week option"
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-a4pt68-MuiButtonBase-root-MuiButton-root"
            role="button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconSchedule css-lw08pq-MuiSvgIcon-root"
              data-testid="appointment_all_availability_previous_week_button"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Navigate to next week option"
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-a4pt68-MuiButtonBase-root-MuiButton-root"
            role="button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconSchedule css-1c40g22-MuiSvgIcon-root"
              data-testid="appointment_all_availability_next_week_button"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth dividerSchedule css-9mgopn-MuiDivider-root"
      />
      <div
        class="css-1lwbda4-MuiStack-root"
        data-testid="appointment_all_availability_schedule_ontainer"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DayAvailability Components FilterHeadingFilled render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="scheduleTimeContainer MuiBox-root css-0"
        >
          <p
            aria-label=""
            class="MuiTypography-root MuiTypography-body1 scheduleTimeTitle css-ahj2mt-MuiTypography-root"
            tabindex="0"
          />
          <div
            class="iconTimeContainer MuiBox-root css-0"
          >
            <button
              aria-label="Navigate to previous week option"
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-a4pt68-MuiButtonBase-root-MuiButton-root"
              role="button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconSchedule css-lw08pq-MuiSvgIcon-root"
                data-testid="appointment_all_availability_previous_week_button"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="Navigate to next week option"
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-a4pt68-MuiButtonBase-root-MuiButton-root"
              role="button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconSchedule css-1c40g22-MuiSvgIcon-root"
                data-testid="appointment_all_availability_next_week_button"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth dividerSchedule css-9mgopn-MuiDivider-root"
        />
        <div
          class="css-1lwbda4-MuiStack-root"
          data-testid="appointment_all_availability_schedule_ontainer"
        />
      </div>
    </div>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="scheduleTimeContainer MuiBox-root css-0"
        >
          <p
            aria-label=""
            class="MuiTypography-root MuiTypography-body1 scheduleTimeTitle css-ahj2mt-MuiTypography-root"
            tabindex="0"
          />
          <div
            class="iconTimeContainer MuiBox-root css-0"
          >
            <button
              aria-label="Navigate to previous week option"
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-a4pt68-MuiButtonBase-root-MuiButton-root"
              role="button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconSchedule css-lw08pq-MuiSvgIcon-root"
                data-testid="appointment_all_availability_previous_week_button"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="Navigate to next week option"
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-a4pt68-MuiButtonBase-root-MuiButton-root"
              role="button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconSchedule css-1c40g22-MuiSvgIcon-root"
                data-testid="appointment_all_availability_next_week_button"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth dividerSchedule css-9mgopn-MuiDivider-root"
        />
        <div
          class="css-1lwbda4-MuiStack-root"
          data-testid="appointment_all_availability_schedule_ontainer"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="scheduleTimeContainer MuiBox-root css-0"
      >
        <p
          aria-label=""
          class="MuiTypography-root MuiTypography-body1 scheduleTimeTitle css-ahj2mt-MuiTypography-root"
          tabindex="0"
        />
        <div
          class="iconTimeContainer MuiBox-root css-0"
        >
          <button
            aria-label="Navigate to previous week option"
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-a4pt68-MuiButtonBase-root-MuiButton-root"
            role="button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconSchedule css-lw08pq-MuiSvgIcon-root"
              data-testid="appointment_all_availability_previous_week_button"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Navigate to next week option"
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-a4pt68-MuiButtonBase-root-MuiButton-root"
            role="button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconSchedule css-1c40g22-MuiSvgIcon-root"
              data-testid="appointment_all_availability_next_week_button"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth dividerSchedule css-9mgopn-MuiDivider-root"
      />
      <div
        class="css-1lwbda4-MuiStack-root"
        data-testid="appointment_all_availability_schedule_ontainer"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
